src/management-system-v2/lib/engines/mqtt-endpoints.ts

```diff
@@ -124,3 +77,92 @@ export async function mqttRequest(
 
   return response;
 }
+
+const collectedDataEntries: Map<
+  string,
+  { onMessageCallback: mqtt.OnMessageCallback; data?: any; clearInterval?: NodeJS.Timeout }
+> =
+  (globalThis as any).collectedDataEntries ||
+  ((globalThis as any).collectedDataEntries = new Map());
+
+export async function collectMqttData<TData>(
+  topic: string,
+  accumulator: (topic?: string, data?: any, previousState?: TData) => TData | undefined,
+  validTopic?: (topic: string) => boolean,
+  staleAfter?: number,
+): Promise<TData | undefined> {
+  let collectedData = collectedDataEntries.get(topic);
+
+  if (!collectedData) {
+    collectedData = {
+      onMessageCallback(messageTopic, message) {
+        if (validTopic ? !validTopic(messageTopic) : messageTopic !== topic) return;
+
+        if (staleAfter) {
+          clearTimeout(collectedData!.clearInterval);
+          collectedData!.clearInterval = setTimeout(() => clearCollectedData(topic), staleAfter);
```

Comment:
Is the name `clearInterval` correct if it is not actually a recurring occurrence

Reply:
yeahp, you're right, that was an old name I forgot to change.
I'll rename it in an upcoming PR.